active/0003-extensible-resource-deployment-observability-improvements.md

```diff
@@ -0,0 +1,311 @@
+
+## Terms and definitions
+
+### Bicep Extension
+
+An API abstraction for the following operations:
+
+- Creating or updating Azure Resources or extensible resources declared in Bicep files or ARM templates;
+- Performing resource-level or extension-level queries that have no side effects through functions in Bicep files or ARM templates, e.g., list secrets of a resource (currently, this is only supported by the az extension).
+
+### Extensible Resource
+
+An Azure data-plane resource or non-Azure resource declared in a Bicep file or an ARM template. Extensible resources are deployed through Bicep extensions.
+
+## Motivation
+
+Users currently encounter a major challenge when deploying extensible resources, as there is inadequate information available when querying the deployments or deployment operation APIs. Key details, such as resource IDs, are not applicable to extensible resources, making it difficult for users to monitor the progress of their deployments. For instance, Azure Portal users attempting to view extensible resource deployment on the deployment details blade will only see an infinite loading animation without any details being displayed. Similarly, Azure PowerShell users tracking deployment progress using the deployment cmdlet and the `-Verbose` switch will receive logs like `VERBOSE: 00:00:00 - Resource '' provisioning status is succeeded`.
+
+To address this issue, the deployments and deployment operations APIs must be updated to include additional information, such as:
+
+- Deployment resource ID
+- Bicep extension alias
+- Bicep extension name
+- Bicep extension version
+- Resource symbolic name
+- Resource type
+- Resource API version
+- Resource identifiers
+
+This additional data would enable users to identify a deployed extensible resource and map it to the corresponding resource definition in a Bicep file or an ARM template, providing users with comprehensive insights.
+
+## Detailed design
+
+### API change for `GET /Microsoft.Resources/deployments/{deploymentId}`
+
+#### Adding `extensions`
+
+The `Microsoft.Resources/deployments` API will be updated to include an `extensions` array in the GET response body. This modification is intended to incorporate information about extensions that are involved in executing the parent deployment and nested deployments.
+
+```diff
+{
+  ...
+  "type": "Microsoft.Resources/deployments",
+  "properties": {
+    ...
+    "providers": [
```

Comment:
Off-topic, but I wonder if anyone actually uses this. Maybe we should consider cutting it to reduce confusion?

Reply:
I removed it from the code sample. I'm not sure why it is there since I inherited this REP. I'm going to leave anything pertaining to this property out of this REP so it focuses on just the extensions/outputResources/operation changes for extensions.